pkg/cache/sql/db/client_test.go

```diff
@@ -0,0 +1,515 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"os"
+	"reflect"
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/assert"
+)
+
+// Mocks for this test are generated with the following command.
+//go:generate mockgen --build_flags=--mod=mod -package db -destination ./db_mocks_test.go github.com/rancher/lasso/pkg/cache/sql/db Rows,Connection,Encryptor,Decryptor,TXClient
+//go:generate mockgen --build_flags=--mod=mod -package db -destination ./transaction_mocks_test.go github.com/rancher/lasso/pkg/cache/sql/db/transaction Stmt,SQLTx
+
+type testStoreObject struct {
+	Id  string
+	Val string
+}
+
+func TestNewClient(t *testing.T) {
+	type testCase struct {
+		description string
+		test        func(t *testing.T)
+	}
+
+	var tests []testCase
+
+	// Tests with shouldEncryptSet to false
+	tests = append(tests, testCase{description: "Query rows with no params, no errors", test: func(t *testing.T) {
+		c := SetupMockConnection(t)
+		e := SetupMockEncryptor(t)
+		d := SetupMockDecryptor(t)
+		expectedClient := Client{
+			conn:      c,
+			encryptor: e,
+			decryptor: d,
+		}
+		client, err := NewClient(c, e, d)
+		assert.Nil(t, err)
+		assert.Equal(t, expectedClient, *client)
```

Comment:
My IDE complains that `expectedClient` doesn't have a `connLock  sync.RWMutex` field

Reply:
I believe its complaining that they're both not pointers and will end up with the exact same locks. I'm not sure why I compared their values. Maybe I mistakenly though it would compare their addresses but it doesn't. Should be gone now.